Bottom: 2ef2e787e3f51dad9e35f40c4d2813d102da670d
Top:    c61f50dfa82d4c1f826805d57e86f0aead006444
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-03-13 13:54:14 +1300

Refresh of extended-to-stream

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index 8a9ce01..93f427a 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -1217,10 +1217,6 @@ class ExtendedToStreamDecorator(CopyStreamResult, StreamSummary, TestControl):
         self.__now = None
         self._started = True
 
-    def stopTestRun(self):
-        super(ExtendedToStreamDecorator, self).stopTestRun()
-        self._started = False
-
     def stopTest(self, test):
         self._tags = self._tags.parent
 
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 90b7ac6..c964f62 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -757,6 +757,14 @@ class TestExtendedToStreamDecorator(TestCase):
               now),
              ('stopTestRun',)], log._events)
 
+    def test_wasSuccessful_after_stopTestRun(self):
+        log = LoggingStreamResult()
+        result = ExtendedToStreamDecorator(log)
+        result.startTestRun()
+        result.status(test_id='foo', test_status='fail')
+        result.stopTestRun()
+        self.assertEqual(False, result.wasSuccessful())
+        
 
 class TestStreamFailFast(TestCase):
